File: ai/src/main/scala/org/pingel/gestalt/core/TransformVertex.java
package org.pingel.gestalt.core;

import org.pingel.util.DirectedGraphVertex;

public class TransformVertex 
implements DirectedGraphVertex<TransformEdge>
{
    public Name name;
    private boolean isStart;
    private boolean isExit;
    
    public TransformVertex(Name name, boolean is, boolean ie)
    {
        this.name = name;
        this.isStart = is;
        this.isExit = ie;
    }
    
    public boolean isStart()
    {
    	return isStart;
    }

    public void isStart(boolean is)
    {
    	isStart = is;
    }
    
    public boolean isExit()
    {
        return isExit;
    }

    public void isExit(boolean ie)
    {
        this.isExit = ie;
    }

    public String toString()
    {
        return name.toString();
    }

    
}

File: ai/src/main/scala/org/pingel/gestalt/core/TransformVertex.scala
package org.pingel.gestalt.core

import org.pingel.util.DirectedGraphVertex

case class TransformVertex(name: Name, var isStart: Boolean, var isExit: Boolean)
extends DirectedGraphVertex[TransformEdge]
{
    def setIsStart(is: Boolean): Unit = {
      isStart = is
    }

    def setIsExit(is: Boolean): Unit = {
      isExit = is
    }

    def toString() = name.toString()
    
}
